src/t_string.c

```diff
@@ -100,7 +102,27 @@ void setGenericCommand(client *c,
         if (getGenericCommand(c) == C_ERR) return;
     }
 
-    found = (lookupKeyWrite(c->db, key) != NULL);
+    robj *existing_value = lookupKeyWrite(c->db, key);
+    found = existing_value != NULL;
+
+    /* Handle the IFEQ conditional check */
+    if (flags & OBJ_SET_IFEQ && found) {
+        if (!(flags & OBJ_SET_GET) && checkType(c, existing_value, OBJ_STRING)) {
```

Comment:
can this happen? i see in the above, we already do this getGenericCommand thing, this will handle all these error path i guess?
```
    if (flags & OBJ_SET_GET) {
        if (getGenericCommand(c) == C_ERR) return;
    }
```

supporting GET and IFEQ look odd to me. GET will get the old value, isn't the value passed in by IFEQ is the same old value?

Reply:
thanks @enjoy-binbin for the review. 

>can this happen?

We currently just want to support Check and Set if the existing value is string. The check for `OBJ_SET_GET` flag prevents us to add additional line of response, when get is supposed to already add response before. 

>supporting GET and IFEQ look odd to me

I feel it's okay to support `GET` with `IFEQ`, in case the user gives an incorrect existing value, and `GET` can help retrieve it. It would also solve the discussion around if `IFEQ` should return the value incase the input doesn't matches the existing value (adding GET flag can always do that now).

@madolson @hpatro what do you guys think on this?